nebula_pytest_bdd/common/nebula_service.py

```diff
@@ -0,0 +1,183 @@
+
+    def set_work_dir(self, work_dir):
+        self.work_dir = work_dir
+
+    def _copy_nebula_conf(self):
+        graph_path = self.build_dir + '/bin'
+        graph_conf_path = self.src_dir + '/conf'
+        storage_path = self.src_dir + '/build/modules/storage/bin'
+        storage_conf_path = self.src_dir + '/modules/storage/conf'
+
+        # graph
+        shutil.copy(graph_path + '/nebula-graphd', self.work_dir + '/bin/')
+        shutil.copy(graph_conf_path + '/nebula-graphd.conf.default',
+                    self.work_dir + '/conf/nebula-graphd.conf')
+        # storage
+        shutil.copy(storage_path + '/nebula-storaged', self.work_dir + '/bin/')
+        shutil.copy(storage_conf_path + '/nebula-storaged.conf.default',
+                    self.work_dir + '/conf/nebula-storaged.conf')
+        # meta
+        shutil.copy(storage_path + '/nebula-metad', self.work_dir + '/bin/')
+        shutil.copy(storage_conf_path + '/nebula-metad.conf.default',
+                    self.work_dir + '/conf/nebula-metad.conf')
+
+        # gflags.json
+        resources_dir = self.work_dir + '/share/resources/'
+        os.makedirs(resources_dir)
+        shutil.copy(self.build_dir + '/../resources/gflags.json', resources_dir)
+
+    def _format_nebula_command(self, name, meta_port, ports, debug_log=True):
+        param_format = "--meta_server_addrs={} --port={} --ws_http_port={} --ws_h2_port={} --heartbeat_interval_secs=1"
+        param = param_format.format("127.0.0.1:" + str(meta_port), ports[0],
+                                    ports[1], ports[2])
+        if name == 'graphd':
+            param += ' --enable_optimizer=true'
+            param += ' --enable_authorize=true'
+        if name == 'storaged':
+            param += ' --raft_heartbeat_interval_secs=30'
+        if debug_log:
+            param += ' --v=4'
+        command = NEBULA_START_COMMAND_FORMAT.format(name, name, param)
+        return command
+
+    def _find_free_port(self):
+        ports = []
+        for i in range(0, 3):
+            with closing(socket.socket(socket.AF_INET,
+                                       socket.SOCK_STREAM)) as s:
+                s.bind(('', 0))
+                s.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, 1)
+                ports.append(s.getsockname()[1])
+        return ports
+
+    def _telnet_port(self, port):
+        with closing(socket.socket(socket.AF_INET, socket.SOCK_STREAM)) as sk:
+            sk.settimeout(1)
+            result = sk.connect_ex(('127.0.0.1', port))
+            return result == 0
+
+    def install(self):
+        if os.path.exists(self.work_dir):
+            shutil.rmtree(self.work_dir)
+        os.mkdir(self.work_dir)
+        print("work directory: " + self.work_dir)
+        os.chdir(self.work_dir)
+        installed_files = ['logs', 'bin', 'conf', 'data', 'pids', 'scripts']
+        for f in installed_files:
+            os.mkdir(self.work_dir + '/' + f)
+        self._copy_nebula_conf()
```

Comment:
From the caller's point of view, after I invoked `install`, Why my working directory is changed?